src/main/java/com/google/devtools/build/lib/rules/cpp/ExtraLinkTimeLibraries.java

```diff
@@ -116,4 +124,32 @@ public BuildLibraryOutput buildLibraries(
     }
     return new BuildLibraryOutput(linkerInputs.build(), runtimeLibraries.build());
   }
+
+  @StarlarkMethod(
+      name = "build_libraries",
+      documented = false,
+      useStarlarkThread = true,
+      parameters = {
+        @Param(name = "ctx", positional = false, named = true),
+        @Param(name = "static_mode", positional = false, named = true),
+        @Param(name = "for_dynamic_library", positional = false, named = true),
+      })
+  public Tuple getBuildLibrariesForStarlark(
+      StarlarkRuleContext starlarkRuleContext,
+      boolean staticMode,
+      boolean forDynamicLibrary,
+      StarlarkThread thread)
+      throws EvalException {
+    CcModule.checkPrivateStarlarkificationAllowlist(thread);
+    try {
+      BuildLibraryOutput buildLibraryOutput =
+          buildLibraries(starlarkRuleContext.getRuleContext(), staticMode, forDynamicLibrary);
+      Depset linkerInputs =
+          Depset.of(CcLinkingContext.LinkerInput.TYPE, buildLibraryOutput.getLinkerInputs());
+      Depset runtimeLibraries = Depset.of(Artifact.TYPE, buildLibraryOutput.getRuntimeLibraries());
+      return Tuple.pair(linkerInputs, runtimeLibraries);
+    } catch (InterruptedException | RuleErrorException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)